hashicorp/terraform-provider-tfe
# feat: Data source for workspace tags in an organization

## Description

This is a new data source is meant to retrieve all the workspace tags for a given organization. With this new data source, we can get the ID, name, and instance count for the organization.

## Testing plan

1. Create a Terraform Cloud/Enterprise organization
2. Create a new workspace with tags (For example, `foo` and `bar`)
3. Create another workspace with similar tags and different tags (For example, `foo` and `test`)
4. Write the necessary Terraform configuration to pull down the tags for a given organization.

```
data "tfe_organization_workspace_tags" "example" {
  organization = "my-org-name"
}
```
5. In the `tags` attribute, we should see something like

```
[
  {
    id = "tag-12345q"
    name = "foo"
    instance_count = "2"
  },
  {
    id = "tag-qwert12"
    name = "bar"
    instance_count = "1"
  },
  {
    id = "tag-qwe333rt12"
    name = "test"
    instance_count = "1"
  },
]
```

Doc Preview:

![image](https://user-images.githubusercontent.com/15823106/213601122-fa159bcc-3f72-4518-b5f6-dea72ebe1fe8.png)

## External links
- [go-tfe documentation](https://pkg.go.dev/github.com/hashicorp/go-tfe#OrganizationTags)
- [API documentation](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/organization-tags#list-tags)
